src/bms/parse/rng.rs

```diff
@@ -20,3 +33,69 @@ impl<const N: usize> Rng for RngMock<N> {
         self.0[N - 1].clone()
     }
 }
+
+/// A random generator for using [`rand`] crate.
+///
+/// [`rand`]: https://crates.io/crates/rand
+/// - Example:
+/// ```rust
+/// use bms_rs::bms::parse::rng::{Rng, RandRng};
+/// use rand::{rngs::StdRng, SeedableRng};
+/// use num::BigUint;
+///
+/// let mut rng = RandRng(StdRng::seed_from_u64(42));
+/// let n = rng.generate(num::range_inclusive(BigUint::from(1u64), BigUint::from(10u64)));
+/// assert!(n >= BigUint::from(1u64) && n <= BigUint::from(10u64));
+/// ```
+#[cfg(feature = "rand")]
+pub struct RandRng<R: rand::RngCore>(pub R);
```

Comment:
```suggestion
pub struct RandRng<R>(pub R);
```
In the best practice, you shouldn't add trait bounds for type parameters in a struct definition. It prevents us from adding features that not depends on `RngCore`.